models/errors.go

```diff
@@ -0,0 +1,32 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrDisconnected = NewRecoverableError(errors.New("disconnected"))
+)
+
+func NewRecoverableError(err error) RecoverableError {
+	return RecoverableError{err}
+}
+
+// RecoverableError is used to signal any types of errors that if encountered
+// could be retried again
+type RecoverableError struct {
+	err error
+}
+
+func (r RecoverableError) Unwrap() error {
+	return r.err
+}
+
+func (r RecoverableError) Error() string {
+	return fmt.Sprintf("recoverable error: %v", r.err)
+}
+
+func IsRecoverableError(err error) bool {
+	return errors.As(err, &RecoverableError{})
```

Comment:
```suggestion
	return errors.Is(err, RecoverableError{})
```

Reply:
This won't work since we defined a custom error type.

Reply:
`errors.Is` would only work if we want to compare error to a specific error value not to a type.

Reply:
Ah, yes, my bad :sweat_smile:

Reply:
So if we would have a recoverable error as:
```
ErrRecoverable = errors.New("receoverable")
```
we could check with `errors.Is` but then that wouldn't work with any error, you could wrap it like: 
```
fmt.Errorf("disconnected: %w", ErrRecoverable) 
```
and that would work, but I find defining a custom type for this case a nicer pattern. But then that makes the `errors.Is` not working.

Reply:
Yeah, good point. I was totally off on this :see_no_evil:

Reply:
yeah no worries, this is confusing API